sdk/servicebus/azure-servicebus/azure/servicebus/_common/_connection_string_parser.py

```diff
@@ -0,0 +1,67 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+try:
+    from urllib.parse import urlparse
+except ImportError:
+    from urlparse import urlparse  # type: ignore
+
+from azure.servicebus.management._models import DictMixin
+
+class ServiceBusConnectionStringProperties(DictMixin):
+    """
+    Properties of a connection string.
+    """
+    def __init__(self, **kwargs):
+        self.fully_qualified_namespace = kwargs.pop('fully_qualified_namespace', None)
+        self.endpoint = kwargs.pop('endpoint', None)
+        self.entity_path = kwargs.pop('entity_path', None)
+        self.shared_access_signature = kwargs.pop('shared_access_signature', None)
+        self.shared_access_key_name = kwargs.pop('shared_access_key_name', None)
+        self.shared_access_key = kwargs.pop('shared_access_key', None)
+
+
+class ServiceBusConnectionStringParser(object):
+    """Parse the connection string.
+
+    :param conn_str: The connection string that has to be parsed.
+    """
+    def __init__(self, conn_str):
+        # type: (str) -> None
+        """
+        :param conn_str: The connection string to parse.
+        :type conn_str: str
+        """
+        self._conn_str = conn_str
+
+    def parse(self):
+        # type() -> ServiceBusConnectionStringProperties
+        """
+        Parse the connection string.
+        """
+        conn_str = self._conn_str.rstrip(";")
+        conn_settings = [s.split("=", 1) for s in conn_str.split(";")]
+        if any(len(tup) != 2 for tup in conn_settings):
+            raise ValueError("Connection string is either blank or malformed.")
+        conn_settings = dict(conn_settings)
+        shared_access_signature = conn_settings.get('SharedAccessSignature')
+        shared_access_key = conn_settings.get('SharedAccessKey')
```

Comment:
do we care whether keys are capital or not?

cuz I see the `_parse_conn_str` in our `_base_handler.py` is ignorant of capitalization, we would call `key.lower()` for key comparison.

Reply:
All the keys other than SharedAccessSignature have a format and casing that is well defined. My recommendation would be to stick to that. I believe the other languages are being case sensitive

Reply:
will change it to lower comparison for SharedAccessSignature